### PR TITLE
Modify rule S1176: Migrate to LayC - Public types, methods and field (API) should be documented

### PR DESCRIPTION
**Pull request by ADarko22**

### CODE REVIEW

File: rules/S1176/php/rule.adoc [hunk 1]
```diff
@@ -1,28 +1,38 @@
+include::../introduction.adoc[]
```
Comment (ADarko22) [inline rules/S1176/php/rule.adoc:1]: It is not implemented in PHP

File: rules/S1176/articles-and-blog-posts.adoc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+* Technical Writer HQ - https://technicalwriterhq.com/documentation/api-documentation/how-to-write-api-documentation/[How to write API documentation]
```
Comment (ADarko22) [inline rules/S1176/articles-and-blog-posts.adoc:1]: I'm not 100% sure if these resources are valid or not

File: rules/S1176/why-is-this-an-issue.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Undocumented APIs pose significant challenges in software development for several reasons:
+
+* *Lack of Clarity:* developers struggling to understand how to use the API correctly. This can lead to misuse and unexpected results.
+
+* *Increased Development Time:* developers spending extra time reading and understanding the source code, which slows down the development process.
+
+* *Error Prone:* developers are more likely to make mistakes that lead to bugs or system crashes when the intent or the error handling os an API is not clear.
```
Comment (petertrr) [inline rules/S1176/why-is-this-an-issue.adoc:7]:
```suggestion
* *Error Prone:* developers are more likely to make mistakes that lead to bugs or system crashes when the intent or the error handling of an API is not clear.
```
typo

File: rules/S1176/java/rule.adoc [hunk 4]
```diff
@@ -1,92 +1,101 @@
...
-* All checked exceptions documented with ``++@throws++``
-* ``++@return++`` present and documented when not ``++void++``.
-* Placeholders like ``++"TODO"++``, ``++"FIXME"++``, ``++"..."++`` should be avoided.
+=== Exceptions
 
 The following public methods and constructors are not taken into account by this rule:
 
 * Getters and setters.
-* Methods overriding another method (usually decorated with ``++@Override++``).
+* Methods overriding another method (usually decorated with `@Override`).
```
Comment (petertrr) [inline rules/S1176/java/rule.adoc:15]:
```suggestion
* Methods overriding another method (usually annotated with `@Override`).
```

File: rules/S1176/java/rule.adoc [hunk 5]
```diff
@@ -1,92 +1,101 @@
...
+* Parameters, using `@param parameterName`.
+* Thrown exceptions, using `@throws exceptionName`.
+* Method return values, using `@return`.
+* Generic types, using `@param <T>`.
+
+Furthermore, the following guidelines should be followed:
+
+* At least 1 line of description.
+* All parameters documented with `@param`, and names should match.
+* All checked exceptions documented with `@throws`
```
Comment (petertrr) [inline rules/S1176/java/rule.adoc:32]:
```suggestion
* All checked exceptions should be documented with `@throws`
```

File: rules/S1176/java/rule.adoc [hunk 6]
```diff
@@ -1,92 +1,101 @@
...
+* `@return` present and documented when not `void`.
```
Comment (petertrr) [inline rules/S1176/java/rule.adoc:33]:
```suggestion
* `@return` present and documented when method return type is not `void`.
```

File: rules/S1176/php/rule.adoc (hunk 1 above)
Comment (petertrr) [reply rules/S1176/php/rule.adoc:1]: I'll double check in our squad if it should be implemented

File: rules/S1176/articles-and-blog-posts.adoc (hunk 2 above)
Comment (petertrr) [reply rules/S1176/articles-and-blog-posts.adoc:1]: I don't see an issue with this, as it's the `blog posts` section

**Review by petertrr — APPROVED** [review]
LGTM, see my few suggestions

**Pull request merged**